### PR TITLE
THORN-2127: rest client 1.1.0

### PR DESCRIPTION
**Pull request by michalszynkiewicz**
- [X] Have you followed the guidelines in our [Contributing](http://wildfly-swarm.io/community/contributing/) document?
- [X] [v2] Have you created a [JIRA](https://issues.jboss.org/browse/THORN) and used it in the commit message?
- [ ] [v4] Have you created a [GitHub Issue](https://github.com/thorntail/thorntail/issues) and used it in the commit message?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/thorntail/thorntail/pulls) for the same issue?
- [X] Have you built the project locally prior to submission with `mvn clean install`?

-----

### CODE REVIEW

File: build-parent/pom.xml [hunk 1]
```diff
@@ -99,7 +99,7 @@
 
     <version.maven.aether>3.2.5</version.maven.aether>
 
-    <version.resteasy>3.0.24.Final</version.resteasy>
+    <version.resteasy>3.6.2.Final</version.resteasy>
```
Comment (Ladicek) [inline build-parent/pom.xml]: Would it work with 3.6.1.Final? That's what WildFly 14.0.1.Final has.

Comment (michalszynkiewicz) [reply build-parent/pom.xml]: I'll try it out. SmallRye is on 3.6.2, that's why I used it.

Comment (Ladicek) [reply build-parent/pom.xml]: Ah OK. Dependency alignment is a nightmare :-(

Comment (michalszynkiewicz) [reply build-parent/pom.xml]: turns out it doesn't work either way, WF11 brings in jaxrs 2.0. I must've run something wrong while testing locally. I will wait with this change for WF14

Comment (Ladicek) [reply build-parent/pom.xml]: If you wish, you can try with the WF14 branch: https://github.com/sberyozkin/thorntail/commits/THORN-2207-wf14 (Just note that we regularly rebase and force-push :-) )

Comment (michalszynkiewicz) [reply build-parent/pom.xml]: I tried it on this branch and it worked, I'll rebase this PR on top of wf14 when it's merged. Thanks!

**Comment by mkouba** [comment]
@michalszynkiewicz the build failed with plenty of MP Rest Client TCK and internal Rest Client test suite errors. Do you know what the cause is?

**Comment by michalszynkiewicz** [comment]
@mkouba the problem is jaxrs api 2.0 leaking in from WF11
I tried on WF14 and it seems not to suffer from this problem, I will rebase this PR when WF14 changes are merged.

**Comment by kenfinnigan** [comment]
@michalszynkiewicz I did some manual merging on this after WF 14 PR was merged but now I'm not sure I did it right.

Could you take a look? Also, I'm not sure if the JAX-RS Spec API is needed anymore either

**Comment by michalszynkiewicz** [comment]
@kenfinnigan thanks!
I reworked it a bit to use RestEasy and JAX-RS versions from WF14

**Comment by kenfinnigan** [comment]
Thanks @michalszynkiewicz, but it might be necessary to completely rebase your PR and force push, or even open a fresh one.

I think my rebase messed up `IncludedTechnology` and `DependencyManagerTest`.

Is that something you could do tomorrow? Sorry for the hassle

**Comment by michalszynkiewicz** [comment]
retest this please

**Comment by michalszynkiewicz** [comment]
@kenfinnigan I minimized the changes :)

**Review by kenfinnigan — APPROVED** [review]

**Pull request merged**